src/common/server.ts

```diff
@@ -83,6 +85,28 @@ async function getRuffVersion(executable: string): Promise<VersionInfo> {
   return { major, minor, patch };
 }
 
+/**
+ * Validate and log executable from given path.
+ */
+async function validateUsingExecutable(executable: string, strategy: string) {
+  try {
+    const ruffVersion = await getRuffVersion(executable);
+    if (!supportsExecutable(ruffVersion)) {
+      var message = `Skip unsupported executable from ${strategy}: ${executable}`;
+      message += `(Reqiuired at least ${versionToString(
+        MINIMUM_SUPPORTED_EXECUTABLE_VERSION,
+      )}, but found ${versionToString(ruffVersion)} instead)`;
+      traceError(message);
```

Comment:
I think we should also display this message to the users using `vscode.window.showErrorMessage`. What do you think?

Reply:
during `findRuffBinaryPath` it may skip several binaries, I think it'd be annoying to send all of them as user facing warnings.

Reply:
Oh, it _is_ for the native server, but we already have a check for that using `MINIMUM_NATIVE_SERVER_VERSION` variable:

https://github.com/astral-sh/ruff-vscode/blob/8f0e0c308d810dbb9d4ff68469db8c9b2109d279/src/common/server.ts#L181-L188

So, this isn't required here.

Reply:
This message shouldn't be required because we already check that later on. Refer to my other comment.

Reply:
🤷‍♂️